sql/core/src/main/scala/org/apache/spark/sql/jdbc/OracleDialect.scala

```diff
@@ -228,8 +228,8 @@ private case class OracleDialect() extends JdbcDialect with SQLConfHelper with N
     extends JdbcSQLQueryBuilder(dialect, options) {
 
     override def build(): String = {
-      val selectStmt = s"SELECT $columnList FROM ${options.tableOrQuery} $tableSampleClause" +
-        s" $whereClause $groupByClause $orderByClause"
+      val selectStmt = s"SELECT $hintClause$columnList FROM ${options.tableOrQuery}" +
```

Comment:
Should we leave a blank between `hintClause` and `columnList`?

Reply:
The reason why I don't add a space here is because if the hint is "", for other JDBCDialects that do not support hints, there will be one more space here than before. But for normal hints, when processing the hint content in `JDBCOptions`, a space has been added at the end.

`s"$value "`

Reply:
The other parts of selectStmt inserts extra space as well. The underlying database can treat these spaces.

Reply:
Ur, @beliefer . I'd prefer not to add a space with a different reason.

The reason is that HINT is very exceptional use cases. I believe 99% queries will not have this hint option. And, we don't want to add an extra space into our SELECT statement here. It's a kind of regression without any benefit.

Reply:
Maybe, we want to inject a space only if HINT exists as an alternative if we want to space, @beliefer . To be clear, I just want to avoid case (1) and not against (3).
```
SELECT c1, c2 FROM t // original
SELECT  c1, c2 FROM t // (1) a normal query with side-effect
SELECT /*+ HINT */c1, c2 FROM t // (2) this PR
SELECT /*+ HINT */ c1, c2 FROM t // (3) Jiaan's suggestion
```

Reply:
Yes, that's my purpose. Only add a space at the end when hint is not empty.

Reply:
@dongjoon-hyun I agree on. Let's check if the hint is not empty.